sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -254,3 +271,197 @@ def _predict_interval(self, fh, X, coverage):
+        self.estimator = estimator
+        self.in_sample_fh = None
+        self.in_sample_X = None
+
+        super().__init__()
+
+    def __repr__(self):
+        return "StatsForecastBackAdapter"
+
+    def fit(self, y, X=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : ndarray
+            Clean time series of shape (t, ).
+        X : typing.Optional[numpy.ndarray], default=None
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        from sklearn.base import clone
+
+        self.in_sample_fh = np.flip(-np.arange(len(y)))
+        estimator = clone(self.estimator)
+        clone_self = self.new()
+        clone_self.estimator = estimator.fit(y=y, X=X)
+
+        return clone_self
+
+    def predict(self, h, X=None, level=None):
+        """Make forecasts.
+
+        Parameters
+        ----------
+        h : int
+            Forecast horizon.
+        X : typing.Optional[numpy.ndarray], default=None
+            Optional exogenous of shape (h, n_x).
+        level : typing.Optional[typing.Tuple[int]], default=None
+            Confidence levels (0-100) for prediction intervals.
+
+        Returns
+        -------
+        y_pred : dict
+            Dictionary with entries mean for point predictions and level_* for
+            probabilistic predictions.
+        """
+        mean = self.estimator.predict(fh=range(1, h + 1), X=X)[:, 0]
+        if level is None:
+            return {"mean": mean}
+
+        level = sorted(level)
+        coverage = [round(1 - (_l / 100), 2) for _l in level]
+
+        pred_int = self.estimator.predict_interval(
+            fh=range(1, h + 1), X=X, coverage=coverage
+        )
+
+        return {
+            "mean": mean,
+            **{
+                f"lo-{_l}": pred_int[("Coverage", c, "lower")].values
+                for c, _l in zip(reversed(coverage), reversed(level))
+            },
+            **{
+                f"hi-{_l}": pred_int[("Coverage", c, "upper")].values
+                for c, _l in zip(coverage, level)
+            },
+        }
+
+    def predict_in_sample(self, level=None):
+        """Access fitted MSTL insample predictions.
+
+        Parameters
+        ----------
+        level : typing.Optional[typing.Tuple[int]]
+            Confidence levels (0-100) for prediction intervals.
+
+        Returns
+        -------
+        y_pred : dict
+            Dictionary with entries mean for point predictions and level_* for
+            probabilistic predictions.
+        """
+        if level and not hasattr(self.estimator, "predict_interval"):
+            raise Exception(
+                "The provided trend_forecaster does not support interval prediction."
+                "Please remove the 'level' parameter or use a different model that"
+                "supports it."
+            )
+
+        fitted = self.estimator.predict(fh=self.in_sample_fh, X=self.in_sample_X)[:, 0]
```

Comment:
Q1. If you are unsure about `predict_interval`, I think you should worry about `predict` as well. Or, don't worry about either.

Q2. Why are you picking only the first column from the predictions?

Reply:
Q1: Gotcha, missed that. Adding check for `predict`

Q2: Output of `predict` gives output of shape [t, 1] which doesn't match expected shape of [t] 
(I'm getting this from `statsforecast` docs, their example has this shape. Not sure if this is an important detail or not)

Reply:
When I read this code, I imeddiately thought may be this is why you're getting just one column in case of multivariate. I'll do some validation on my end to give an updated comment, but won't have time before weekend for work pressure.

Reply:
Ok, if this helps, here is the code that led me to believe `MSTL` only takes univariate: 
https://github.com/Nixtla/statsforecast/blob/main/statsforecast/mstl.py, L28-L29

Reply:
if only univariate is supported, the tag `scitype:y` should be set to `"univariate"` - no extra logic to deal with it is needed

Reply:
`sktime` will per default broadcast across columns